### ISSUE TITLE
Only 6 of 16 settings available

### ISSUE THREAD

**Issue by dallas7** [schomery/privacy-settings#103]
Win7 Home SP2 x64

Thank you for Privacy Settings.  I've long used it with success and satisfaction in earlier versions of Firefox, currently 52.5.2 ESR and your 0.2.6 that remains as my primary browser for the time being. 

Now in a "discovery phase" with 57.0.2 and 0.3.2 I find only these settings are available:
- network.networkPredictionEnabled
- network.peerConnectionEnabled
- network.webRTCIPHandlingPolicy
- services.passwordSavingEnabled
- websites.hyperlinkAuditingEnabled
- websites.referrersEnabled
which are ten short of the 16 detailed in the extension "More" about:addons page.

Nor does the extension drop-down panel resemble the screen shots on your Mozilla Add-ons page.

This is a report, not a complaint.  Perhaps I don't understand the new browser/extension thing or you will have a fix forthcoming.

Best regards.

**Comment by TerkiKerel** [comment]
This project can not be maintained anymore... I could not get any response from simplest messages here..

I am happy wtih ghacks userjs project. I recommend you this:
https://github.com/ghacksuserjs/ghacks-user.js

**Comment by dallas7** [comment]
> This project can not be maintained anymore - I could not get any response

You don't know that.  Jeremy has four other projects taking up his efforts as well and it's certain he won't snap to attention just for you.

I've been RSSing Martin's content for many years and his gHacks user.js is very good but it's just some one else's user.js which were a dime a dozen (including the one's I've built myself ever since Mozilla implemented that).  It loads at browser startup, maintenance/upgrade is tedious and it's a pain to troubleshoot for the inevitable broken sites.  Which is why the all the public custom user.js files fell out of favor long ago.  

Without an interface I loose control and I demand to be in control; Jeremy's extension allowed for that, to toggle privacy targeted settings in real time without restarts.

Thanks for the recommendation anyhow.

**Comment by Thorin-Oakenpants** [comment]
Hi guys (and gals)

The `ghacks user.js` was not written by Martin, just originally started/articled there, FYI. There are so many settings now that it can be problematic (more on that later). The maintenance is done, so no need to worry about that (except the end user should read the changelogs per release - however, this is the same as keeping abreast of changes within an extension). The `ghacks user.js` is rather comprehensive and is a template - so it requires users to read and understand it (problematic in a sense that it takes a lot of reading to fully understand all the settings, and its complex and no one solution fits all). And it has drawbacks (eg changes are only applied on start, clearing or resetting prefs to default values still has to be done manually, fiddling around in the user.js is not everyone's cup of tea, changes in real time need to be done from about:config). But it IS what it is and is a valid technique (as is locking prefs via cfg) - once done/modified, it can almost be forgotten. Any shortfalls have solutions - it just may not be a solution you LIKE, but some people do - going to about:config is quicker for me than flicking open a extension panel. Making changes/looking things up in a single searchable js file is quicker for me, including reference links etc. Each to their own.

An extension such as @schomery 's is brilliant and welcome and mitigates some of those shortfalls some people have. And for the average user, it's great - can still be a little daunting (so many options, same as the user.js), but simplified and streamlined. Here's hoping that Mozilla open up some more settings under the Privacy API

I'm trying to find a meta ticket but so far no luck
- https://bugzilla.mozilla.org/show_bug.cgi?id=1397611 - extensions can now access `privacy.resistFingerprinting`
- https://bugzilla.mozilla.org/show_bug.cgi?id=1409045 -  extensions can control `privacy.firstparty.isolate`

Bringing this up with the Mozilla guys who work on the Tor Uplift guys is an option - ethan, tom ritter etc, but I suspect because so many prefs are involved, and the potential clash with resist.Fingerprinting, that little would be actioned.

tl;dr: Rubbishing the use of a user.js is not helpful/constructive (and likewise belittling an extension due to limitations - both methods have pros and cons), nor is bitching about @schomery 's lack of reply - he's a human devoting his free time, and I have nothing but respect for him.

Chill out guys :)

**Comment by dallas7** [comment]
> is not everyone's cup of tea - is a valid technique

Having used Mozilla as my primary browser since it was Phoenix 0.something in Windows 2000 I can agree that user.js is indeed valid and I believe when I said it "fallen out of favor" is the same as your "not every one's cup of tea."

> and likewise belittling an extension due to limitations

My OP:  "This is a report, not a complaint."
 
> rather comprehensive
> requires users to read and understand it
> it has drawbacks
> clearing or resetting prefs ... has to be done manually

As I said, tedious.  Granted, ghacks user.js itself is a real nice piece of work for those who want to deal with it.  For the record, I have donated to Martin here and there and he's whitelisted in my blocker and tracker extensions/settings.

> once done/modified, it can almost be forgotten

Anything can almost be forgotten (even Privacy Settings).  Until they have to be unforgotten.

> (user.js) may not be a solution you LIKE

True, not a solution I like.  For at least a decade.

> Rubbishing the use of a user.js is not helpful/constructive

Yet you spend a 200+ word paragraph doing just that.  Much much better than my some one else's dime a dozen wrap up.  Thanks.  (Cyber ages ago there were forum threads where folks would contest who has the bestest user.js there ever is.  You shudda seen the rubbishing!)

> extension such as schomery 's is brilliant and welcome

That's why I first thing gave thanks for it up there in my OP.  Hopefully I'll run an extensions update in the near future in my 57 test profile and the next 0.3 will present all the 16 settings (or more) he declares in Add0n and about:preferences More and not the six I reported.  And as some seem to have reviewed about over at the Moz Add-ons site. 

Unsubscibed.

FYI:
ttps://www.ghacks.net/2017/12/29/automation-comes-to-the-ghacks-user-js-configuration-for-firefox/
